src/axom/inlet/docs/sphinx/advanced_types.rst

```diff
@@ -65,51 +115,44 @@ for these associative arrays.
-an instance of type ``T`` with its members populated with the corresponding fields in the input file.
-For the simple ``Mesh`` example whose schema is defined above, the specialization might look like:
-
-.. literalinclude:: ../../examples/user_defined_type.cpp
-   :start-after: _inlet_userdef_simple_frominlet_start
-   :end-before: _inlet_userdef_simple_frominlet_end
-   :language: C++
+As with the schema definition, the ``FromInlet`` specialization for a user-defined type will work for both
+single instances of the type and arrays of the type.
 
-The ``Table::operator[]`` is used to extract fields from the input file and is automatically converted to
-the correct member variable types when the function's return value is constructed.  This conversion does
-not happen automatically for user-defined types.  If a ``Mesh`` object as defined above is located at the
-path "mesh" within the input file, it can be retrieved as follows:
+To retrieve an array of structs as a contiguous array of user-defined type, use ``std::vector``:
 
 .. code-block:: C++
 
-  Mesh mesh = inlet["mesh"].get<Mesh>();
+  auto ops = base["operators"].get<std::vector<Operator>>();
 
-As with the schema definition, the ``FromInlet`` specialization for a user-defined type will work for both
-single instances of the type and arrays of the type.
-
-Arrays in Inlet are implemented as ``std::unordered_map<int, T>``, so retrieving an array of user-defined
-types is as follows, in this case for an array of the ``BoundaryCondition`` struct:
+Some input file languages support non-contiguous array indexing, so you can also retrieve arrays 
+as ``std::unordered_map<int, T>``:
 
 .. code-block:: C++
 
-  auto mesh = inlet["bcs"].get<std::unordered_map<int, BoundaryCondition>>();
+  auto ops = inlet["operators"].get<std::unordered_map<int, Operator>>();
+
+.. note::
+  If a non-contiguous array is retrieved as a (contiguous) ``std::vector``, the elements will be
+  ordered by increasing index.
 
 String-keyed dictionaries are implemented as ``std::unordered_map<std::string, T>`` and can be retrieved
 in the same way as the array above.  For dictionaries with a mix of string and integer keys, the
 ``inlet::VariantKey`` type can be used, namely, by retrieving a ``std::unordered_map<inlet::VariantKey, T>``.
 
-Adding Functions to a Schema
-----------------------------
+*********
+Functions
+*********
 
 For input file types that support functions, e.g., Lua, functions can also be read from the input file
-into a ``std::function``, the wrapper for callables provided by the C++ standard library.  This is accomplished
-by calling ``addFunction`` on an Inlet or Table object.
+into a ``std::function``, the wrapper for callables provided by the C++ standard library.
+
+Defining And Storing
+--------------------
+
+This is accomplished by calling ``addFunction`` on an Inlet or Table object.
 
 Consider the following Lua function that accepts a vector and returns a double:
```

Comment:
Please clarify that `vector` is a spatial 3-vector (rather than an `std::vector`) which was just used in the previous subsection